src/hera/task.py

```diff
@@ -614,10 +609,11 @@ def get_parallel_items(self) -> List[_Item]:
         for func_param in self.func_params:
             item = {}
             for k, v in func_param.items():
-                if isinstance(v, BaseModel):
-                    item[k] = v.json()
-                else:
-                    item[k] = json.dumps(v)  # type: ignore
+                v_json = encode_json(v)
+                try:
+                    item[k] = json.loads(v_json)
+                except json.JSONDecodeError:
+                    item[k] = v_json
```

Comment:
Yea! I was wondering the same thing about using `BaseModel.dict()` as well. I was unsure about how pydantic would behave if we had a model that contains other models in it. I did the loads/dumps routine in order to ensure the resulting object was 100% native json-compatible types. If `.dict()` can make the same guarantee then I definitely think you're way is much nicer!

In terms of the argo side question, I'm not 100% sure. In the argo spec, `withItems` expects a list of maps, so with a single item being submitted I assume argo would see see a list of maps, just the list only contains a single item.

Reply:
The `withItems` field is very finicky, actually... which is why I was asking about the Argo side. The `_Item` was introduced to overcome an open problem in OpenAPI SDK generation. Any chance you have a cluster available to test these changes in? Specifically, the hello world and parallel diamonds example - those are the two core ones I tested when `_Item` was introduced to makes sure things are serialized properly. Alternatively, I can help with this as well as testing can be a bit time consuming since it's best to assemble some tasks with a variety of parameters, such as a string, an integer, and a Pydantic model to test serialization in the single task and parallel task case.

Reply:
I will test with those workflows tomorrow morning!

I will say however that will this change I have been able to submit workloads to my cluster that have very complex `withItems` (strings, ints and dicts) and combinations of multiple and single `func_params`.

Reply:
Oh, that sounds amazing! Thank you for already testing that 🙂 I should also say thank you for using Hera ❤️ Very excited to hear that you are using it to submit fairly complex workflows! If you have the chance then, I guess testing the single pod/single task workflow would be sufficient. At @dynotx we had some serialization issues when submitting `pandas.DataFrame` queries through configs and we had to futz around with the strings that are submitted, which caused problems/differences between the way we submitted single vs. parallel tasks, but we reverted to the current implementation and used it differently 🤷🏻 want to ensure the same thing does not happen here.